### PR TITLE
Back 1773 upgrade node background task to support redis v3

### PR DESCRIPTION
**Pull request by mjsalinger**
MIgraded from node_redis to ioredis.  Changed create methods to include a redisOptions instead of binding to host/port (for sentinel, etc.).  

Added callbacks to all shutdowns.

### CODE REVIEW

File: lib/blacklist.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ var dataStore = require("./data_store"),
  * @type {Function}
  */
 var Blacklist = exports.Blacklist = function(options, callback) {     // Optional callback for auth operation test case
-  // Call to EventEmitter constructor to intitialize arguments
+  // Call to EventEmitter tructor to intitialize arguments
```
Comment (sjl2024) [inline lib/blacklist.js]: comment typo!

File: lib/redis_client.js [hunk 2]
```diff
@@ -645,9 +643,10 @@ exports.initialize = function(options, callback) {
   /**
    * Shut down the redis client
    */
-  RedisClient.prototype.shutdown = function() {
+  RedisClient.prototype.shutdown = function(callback) {
     this.client.removeAllListeners();
-    this.client.end();
+    this.client.disconnect();
+    callback();
```
Comment (sjl2024) [inline lib/redis_client.js]: Are the two methods above guaranteed to be synchronous?

File: lib/task_client.js [hunk 3]
```diff
@@ -189,24 +190,42 @@ exports.initialize = function(options) {
...
-      that.taskLimit.shutdown();
+      shutdownFunctions.push(that.taskLimit.shutdown.bind(that.taskLimit));
     }
-    that.notificationBus.shutdown();
-    that.removeAllListeners();
+
+    async.parallel(shutdownFunctions, function(err) {
+      // Don't worry about errors
+      that.removeAllListeners();
+      if (typeof callback === 'function') {
```
Comment (sjl2024) [inline lib/task_client.js]: Not actually a problem but more of a question for my own understanding... in what circumstances could `callback !== function`?

File: package.json [hunk 4]
```diff
@@ -29,8 +29,9 @@
     "async": "0.2.10",
     "lodash.isnil": "4.0.0",
     "precond": "0.2.2",
-    "redis": "0.8.6",
-    "uuid": "^3.0.0"
+    "ioredis":"2.4.2",
+    "hiredis":"0.5.0",
```
Comment (sjl2024) [inline package.json]: Seemingly-unused dependency

File: lib/task_client.js (hunk 3 above)
Comment (mjsalinger) [inline lib/task_client.js]: If the user just called shutdown()

File: package.json (hunk 4 above)
Comment (mjsalinger) [inline package.json]: ioredis uses hiredis and requires you to include it in your package.

**Comment by sjl2024** [comment]
LGTM

**Pull request merged**